NitroxTest/Serialization/WorldPersistenceTest.cs

```diff
@@ -0,0 +1,426 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using NitroxModel.Core;
+using NitroxModel.DataStructures;
+using NitroxModel.DataStructures.GameLogic;
+using NitroxModel.DataStructures.Util;
+using NitroxServer.GameLogic;
+using NitroxServer.GameLogic.Bases;
+using NitroxServer.GameLogic.Entities;
+using NitroxServer.GameLogic.Items;
+using NitroxServer.GameLogic.Players;
+using NitroxServer.GameLogic.Unlockables;
+using NitroxServer.GameLogic.Vehicles;
+using NitroxServer.Serialization;
+using NitroxServer.Serialization.World;
+using NitroxServer_Subnautica;
+
+namespace NitroxTest.Serialization
+{
+    [TestClass]
+    public class WorldPersistenceTest
+    {
+        private string tempSaveFilePath;
+        private WorldPersistence worldPersistence;
+        private World world;
+
+        private ServerConfig serverConfig;
+        private IServerSerializer[] serverSerializers;
+
+        private PersistedWorldData worldData;
+        private PersistedWorldData[] worldsDataAfter;
+
+        [TestInitialize]
+        public void Setup()
+        {
+            NitroxServiceLocator.InitializeDependencyContainer(new SubnauticaServerAutoFacRegistrar(), new TestAutoFacRegistrar());
+            NitroxServiceLocator.BeginNewLifetimeScope();
+
+            tempSaveFilePath = Path.Combine(Path.GetTempPath(), "NitroxTestTempDir");
```

Comment:
Just a suggestion to maybe use a mock file system so that we don't actually have to save to disk?

Reply:
I can changed it to a stream.

Reply:
Now that I think about it to test the save and load function of the `WorldPersistence.cs` we can't use bytestream or mock filesystem.

Reply:
Looks like all the required APIs used by `LoadFromFile` do accept `Stream` so with a tiny bit of refactoring (moving code into a new `LoadFromStream`) you can make it testable by using `MemoryStream`.

Reply:
For coverage and "don't edit the code to suit the test" I would leave it as it is.